### PR TITLE
can search for users and send messages to them

### PR DESCRIPTION
**Pull request by tsbolty**
### What this PR does (required):
- Can search for users on messaging page and send conversations to them.

### Screenshots / Videos (required):
image

### Any information needed to test this feature (required):
- Search for users in the database

### Any issues with the current functionality (optional):
- No known issues

### CODE REVIEW

File: client/src/context/conversationContext.tsx [hunk 1]
```diff
@@ -1,27 +1,49 @@
...
+        _id: '',
+        profilePic: '',
+        username: '',
+      });
       getOneConvo(friendId)
-        .then((res: Message[]) => {
-          if (res) {
-            // MAY NEED TO MAP THROUGH DATA TO GET CORRECT DATA STRUCTURE BEFORE SETTING STATE WITH IT
+        .then((res) => {
+          console.log(res);
```
Comment (rajivtitus) [inline client/src/context/conversationContext.tsx]: Please remove

File: client/src/components/Search/Search.tsx [hunk 2]
```diff
@@ -68,6 +70,7 @@ const Search = ({ search, handleChange }: Props): JSX.Element => {
         getOptionLabel={(option) => option.username}
         options={options}
         loading={loading}
+        onChange={(e, value: any) => value && value.id && setFriendId(value.id)}
```
Comment (rajivtitus) [inline client/src/components/Search/Search.tsx]: Can we use the correct type for value here? :)

File: client/src/context/conversationContext.tsx (hunk 1 above)
Comment (tsbolty) [reply client/src/context/conversationContext.tsx]: Fixed.

File: client/src/components/Search/Search.tsx (hunk 2 above)
Comment (tsbolty) [reply client/src/components/Search/Search.tsx]: I tried about 50 different things and asked Brian to look at this. Unfortunately nothing we tried provided a solution. However, this value comes from a database response that is checking the data type before it gets to this point. The issue here is that the Autocomplete API from Material-UI takes the object, but also adds optional "string" or "null" values to it. So the id could not be guaranteed. I tried everything to get around this.

Comment (tsbolty) [reply client/src/components/Search/Search.tsx]: Nevermind. Crushed it! Haha. I think it's ready to go now.

**Review by brian-glovebox — COMMENTED** [review]
I am glad you figured out the right way to work around the any type issue. Great work bud!

**Comment by tsbolty** [comment]
> I am glad you figured out the right way to work around the any type issue. Great work bud!

super! So is this approved?

**Pull request merged**